aesm-client/src/imp/aesm_protobuf/mod.rs

```diff
@@ -26,11 +26,14 @@ impl AesmClient {
         #[cfg(not(target_env = "sgx"))]
         let _ = sock.set_read_timeout(req.get_timeout().map(|t| Duration::from_micros(t as _)))?;
 
-        let req_bytes = req
-            .into()
-            .write_to_bytes()
+        let mut req_bytes = vec![0u8; 4];
+        let mut cursor = Cursor::new(&mut req_bytes);
+        cursor.set_position(4);
```

Comment:
You don't need a cursor, [`impl io::Write for Vec<u8>` appends to the vector](https://doc.rust-lang.org/1.42.0/std/io/trait.Write.html#impl-Write-9).